python-sdk/src/astro/sql/table.py

```diff
@@ -125,6 +122,42 @@ def name(self, value: str) -> None:
             self._name = value
             self.temp = False
 
+
+@define(slots=False)
+class TempTable(BaseTable):
+    """
+    Internal class to represent a Temporary table
+
+    :sphinx-autoapi-skip:
+    """
+
+    temp: bool = field(default=True)
```

Comment:
```suggestion
    temp: ClassVar[bool] = True
```

(And then along with a matching change to False in Table I think you can leave it slotted too?)

Reply:
Didn't work unfortunately -- 
```
ImportError while loading conftest '/Users/kaxilnaik/Documents/GitHub/astronomer/astro-sdk/python-sdk/conftest.py'.
conftest.py:14: in <module>
    from astro.constants import Database, FileLocation, FileType
src/astro/__init__.py:17: in <module>
    import astro.sql  # noqa: F401
src/astro/sql/__init__.py:4: in <module>
    from astro.sql.operators.append import AppendOperator, append
src/astro/sql/operators/append.py:8: in <module>
    from astro.databases import create_database
src/astro/databases/__init__.py:5: in <module>
    from astro.databases.base import BaseDatabase
src/astro/databases/base.py:20: in <module>
    from astro.sql.table import Metadata, Table
src/astro/sql/table.py:139: in <module>
    class Table(BaseTable, Dataset):
E   TypeError: multiple bases have instance lay-out conflict
```

Reply:
>However, [it’s not possible](https://docs.python.org/3/reference/datamodel.html#notes-on-using-slots) to inherit from more than one class that has attributes in __slots__ (you will get an TypeError: multiple bases have instance lay-out conflict).

So probably since `airflow.Dataset` has slots and even `BaseTable` has slot It doesn't work :shrug:

https://www.attrs.org/en/stable/glossary.html?highlight=multiple%20bases%20have%20instance%20lay-out%20conflict#term-slotted-classes